include/perf_ringbuffer.hpp

```diff
@@ -7,26 +7,28 @@
 
 #include "perf.hpp"
 
-typedef struct RingBuffer {
-  const char *start;
-  unsigned long offset;
-  size_t mask;
-  size_t meta_size; // size of the metadata page
-  size_t data_size; // size of data pages
-  size_t size;      // size of the entire region (data + metadata)
-  struct perf_event_mmap_page *region;
-  bool is_mirrored;
```

Comment:
interesting, so the mirroring never accounts for more pinned memory

Reply:
Yes, I checked in the kernel source.
cf. https://github.com/DataDog/ddprof/pull/118/commits/fc3a2c1fd4e167b868478d56d2e97ed716e9943b